internal/clients/api_client.go

```diff
@@ -360,3 +185,123 @@ func (a *ApiClient) ClusterID(ctx context.Context) (*string, diag.Diagnostics) {
 	})
 	return nil, diags
 }
+
+func newEsApiClient(d *schema.ResourceData, key string, version string) (*ApiClient, diag.Diagnostics) {
+	var diags diag.Diagnostics
+	config := elasticsearch.Config{}
+	config.Header = http.Header{"User-Agent": []string{fmt.Sprintf("elasticstack-terraform-provider/%s", version)}}
+
+	if esConn, ok := d.GetOk(key); ok {
+		// if defined, then we only have a single entry
+		if es := esConn.([]interface{})[0]; es != nil {
+			esConfig := es.(map[string]interface{})
+
+			if username, ok := esConfig["username"]; ok {
+				config.Username = username.(string)
+			}
+			if password, ok := esConfig["password"]; ok {
+				config.Password = password.(string)
+			}
+			if apikey, ok := esConfig["api_key"]; ok {
+				config.APIKey = apikey.(string)
+			}
+
+			// default endpoints taken from Env if set
+			if endpoints := os.Getenv("ELASTICSEARCH_ENDPOINTS"); endpoints != "" {
```

Comment:
We should only be looking at this environment variable in the root provider case. Probably just need an argument here to say `useEnvAsDefault` or similar.

Reply:
Cheers, I wasn't sure if it was fine as a default for the overrides so I took them out.

I saw env vars for username and password used for `NewAcceptanceTestingClient`, should I bring those in too or those are only supported for testing?